src/main/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingImpl.java

```diff
@@ -465,7 +466,24 @@ public Set<SolutionMapping> translateSolutionMappingFromGlobal( final SolutionMa
 		}
 		return results;
 	}
-	
+
+	@Override
+	public boolean isEquivalenceOnly() {
+		for (final Triple m : getMappings(Node.ANY, Node.ANY, Node.ANY)){
```

Comment:
Calling the `getMappings` function is very inefficient, in particular if you call it with all three positions unbound. The issue is that this function then simply copies the complete graph into another `Set` and returns that. (But, of course, that's not your fault!)  In fact, we should get rid of this function in this class. In the meantime, however, it is better not to use it. Instead, the iterator-based approach as that function uses internally can be used directly here.